### PR TITLE
Merge main into 8.16 branch

### PR DESCRIPTION
**Pull request by rogercoll**

### CODE REVIEW

File: resources/kubernetes/operator/helm/values.yaml [hunk 1]
```diff
@@ -852,7 +873,7 @@ collectors:
             processors:
               - elasticinframetrics
               - batch
-              - k8sattributes
+              - k8sattributes/ecs
               - resourcedetection/system
```
Comment (eedugon) [inline resources/kubernetes/operator/helm/values.yaml:877]: @rogercoll : these changes are not related with the `autoGenerateCert` change I did.
Just FYI.

**Review by eedugon — APPROVED** [review]
LGTM, I have left a comment to double check that the other changes in values.yaml are expected.

File: resources/kubernetes/operator/helm/values.yaml (hunk 1 above)
Comment (rogercoll) [reply resources/kubernetes/operator/helm/values.yaml:877]: You are right, those changes were already merged. I think this is an issue with how GitHub shows the diffs, see the "two dot" difference instead: https://github.com/elastic/opentelemetry/compare/8.16..main
